library/HTMLPurifier/URIFilter/SafeIframe.php

```diff
@@ -56,12 +56,21 @@ public function filter(&$uri, $config, $context)
         if (!($token && $token->name == 'iframe')) {
             return true;
         }
+
         // check if we actually have some whitelists enabled
-        if ($this->regexp === null) {
+        $safeHosts = $config->get('URI.SafeIframeHosts');
+        if ($this->regexp === null && $safeHosts === null) {
             return false;
         }
         // actually check the whitelists
-        return preg_match($this->regexp, $uri->toString());
+        if ($this->regexp !== null && !preg_match($this->regexp, $uri->toString())) {
+            return false;
+        }
+        // check if the host is in a whitelist for safe iframe hosts
+        if ($safeHosts !== null && !isset($safeHosts[$uri->host])) {
+            return false;
+        }
```

Comment:
You know what, I was wrong, disjunction actually seems right 🤣  Because it's a whitelist. So you can whitelist some urls with regex and also whitelist some hosts with safe host...

Reply:
No problem. I changed it back. Thanks.